comp/checks/winregistry/impl/winregistryimpl.go

```diff
@@ -0,0 +1,315 @@
+type WindowsRegistryCheck struct {
+	core.CheckBase
+	metrics.Gauge
+	senderManager           sender.SenderManager
+	sender                  sender.Sender
+	logsComponent           agent.Component
+	log                     log.Component
+	registryKeys            []registryKey
+	registryDelegate        registryDelegate
+	integrationLogsDelegate *integrationLogsRegistryDelegate
+}
+
+// Configure configures the check
+func (c *WindowsRegistryCheck) Configure(senderManager sender.SenderManager, integrationConfigDigest uint64, data integration.Data, initConfig integration.Data, source string) error {
+	c.senderManager = senderManager
+	c.BuildID(integrationConfigDigest, data, initConfig)
+	err := c.CommonConfigure(senderManager, integrationConfigDigest, initConfig, data, source)
+	if err != nil {
+		return err
+	}
+
+	var initCfg checkInitCfg
+	if err := yaml.Unmarshal(initConfig, &initCfg); err != nil {
+		return err
+	}
+
+	var conf checkCfg
+	if err := yaml.Unmarshal(data, &conf); err != nil {
+		return err
+	}
+
+	var sendOnStart, sendOnStartSet bool
+	if sendOnStart, sendOnStartSet = conf.SendOnStart.Get(); !sendOnStartSet {
+		if initSendOnStart, initSendOnStartSet := initCfg.SendOnStart.Get(); initSendOnStartSet {
+			sendOnStart = initSendOnStart
+		} else {
+			sendOnStart = false
+		}
+	}
+
+	hiveMap := map[string]registry.Key{
+		"HKLM":               registry.LOCAL_MACHINE,
+		"HKEY_LOCAL_MACHINE": registry.LOCAL_MACHINE,
+		"HKU":                registry.USERS,
+		"HKEY_USERS":         registry.USERS,
+		"HKEY_CLASSES_ROOT":  registry.CLASSES_ROOT,
+		"HKCR":               registry.CLASSES_ROOT,
+	}
+
+	for regKey, regKeyConfig := range conf.RegistryKeys {
+		splitKeypath := strings.SplitN(regKey, "\\", 2)
+		if len(splitKeypath) != 2 {
+			return fmt.Errorf("the key %s is too short to be a valid key", regKey)
+		}
+
+		if len(regKeyConfig.Name) == 0 {
+			c.log.Warnf("the key %s does not have a metric name, skipping", regKey)
+			continue
+		}
+
+		if hive, found := hiveMap[splitKeypath[0]]; found {
+			regValues := make(map[string]registryValueCfg)
+			for valueName, regValueCfg := range regKeyConfig.RegistryValues {
+				if len(regValueCfg.Name) == 0 {
+					c.log.Warnf("the subkey %s of %s does not have a metric name, skipping", valueName, regKey)
+				} else {
+					regValues[valueName] = regValueCfg
+				}
+			}
+			c.registryKeys = append(c.registryKeys, registryKey{
+				name:            regKeyConfig.Name,
+				hive:            hive,
+				originalKeyPath: regKey,
+				keyPath:         splitKeypath[1],
+				registryValues:  regValues,
+			})
+		} else {
+			return fmt.Errorf("unknown hive %s", splitKeypath[0])
+		}
+	}
+
+	c.sender, err = c.GetSender()
+	if err != nil {
+		return fmt.Errorf("failed to retrieve a sender for check %s: %s", string(c.ID()), err)
+	}
+	c.sender.FinalizeCheckServiceTag()
+
+	c.integrationLogsDelegate = &integrationLogsRegistryDelegate{
```

Comment:
Looks like c.integrationLogsDelegate is used only for muted field and not for actual delegations